ArduCopter/sensors.cpp

```diff
@@ -399,7 +399,7 @@ void Copter::update_sensor_status_flags(void)
     if (!g.compass_enabled || !compass.healthy() || !ahrs.use_compass()) {
         control_sensors_health &= ~MAV_SYS_STATUS_SENSOR_3D_MAG;
     }
-    if (gps.status() == AP_GPS::NO_GPS) {
+    if (!gps.is_healthy()) {
```

Comment:
Any reason why Copter and Sub are different from Plane and Rover?

Reply:
Because @rmackay9 thinks healthy should only indicate if the sensor is talking, rather then if it's useful for nav reasons. This is basically just preserving the existing behavior.

I think copter and sub are wrong, but thats just my opinion. (There is also a bit of a concern if copter users would suddenly be seeing lots of bad gps health warnings/panicking (although again I think it's a fair warning and would help them know that they couldn't use a GPS flight mode))

Reply:
OK, as we keep existing behavior I'll let it go, I just hate when, for the same thing, vehicles do it differently - a user using Plane and Copter will be confused by this.

I'll add that I agree with @rmackay9, I think this reports the health of the sensors, not if they can be used or not. For me, it would make more sense that when GPS can't be used (because it doesn't have at least 3D fix), that the *enabled* bit wasn't set - but it could still be healthy.